node/init.go

```diff
@@ -50,13 +50,13 @@ func Init(path string, tp node.Type, options ...node.Option) error {
 		return err
 	}
 
-	if sets.Cfg == nil {
+	if sets.cfg == nil {
 		return errors.New("configuration is missing for the node's initialisation")
 	}
 
 	cfgPath := configPath(path)
 	if !utils.Exists(cfgPath) {
-		err = node.SaveConfig(cfgPath, sets.Cfg)
+		err = SaveConfig(cfgPath, sets.cfg)
```

Comment:
So something to note here is that I tried: 

```
celestia full init --core.ip https://rpc-mamaki.pops.one --keyring.accname something --core.grpc.port 9090
```

and it didn't persist the core config values i passed to the config on init. 

It also didn't persist the values on start.